boards/nano33ble/src/test/log_test.rs

```diff
@@ -0,0 +1,659 @@
+        }
+
+        self.wait();
+    }
+
+    fn sync_done(&self, error: ReturnCode) {
+        if error == ReturnCode::SUCCESS {
+            debug!(
+                "SYNC DONE: READ OFFSET: {:?} / WRITE OFFSET: {:?}",
+                self.log.next_read_entry_id(),
+                self.log.log_end()
+            );
+        } else {
+            panic!("Sync failed: {:?}", error);
+        }
+
+        self.next_op();
+        self.run();
+    }
+
+    fn erase_done(&self, error: ReturnCode) {
+        match error {
+            ReturnCode::SUCCESS => {
+                // Reset test state.
+                self.op_index.set(0);
+                self.op_start.set(true);
+                self.read_val.set(0);
+                self.write_val.set(0);
+
+                // Make sure that flash has been erased.
+                for i in 0..TEST_LOG.len() {
+                    if TEST_LOG[i] != 0xFF {
+                        panic!(
+                            "Log not properly erased, read {} at byte {}. SUMMARY: {:?}",
+                            TEST_LOG[i],
+                            i,
+                            &TEST_LOG[i..i + 8]
+                        );
+                    }
+                }
+
+                // Make sure that a read on an empty log fails normally.
+                self.buffer.take().map(move |buffer| {
+                    if let Err((error, original_buffer)) = self.log.read(buffer, BUFFER_LEN) {
+                        self.buffer
+                            .replace(original_buffer.expect("No buffer returned in error!"));
+                        match error {
+                            ReturnCode::FAIL => (),
+                            ReturnCode::EBUSY => {
+                                self.wait();
+                            }
+                            _ => panic!("Read on empty log did not fail as expected: {:?}", error),
+                        }
+                    } else {
+                        panic!("Read on empty log succeeded! (it shouldn't)");
+                    }
+                });
+
+                // Move to next operation.
+                debug!("Log Storage erased");
+                self.state.set(TestState::Operate);
+                self.run();
+            }
+            ReturnCode::EBUSY => {
+                // Flash busy, try again.
+                self.wait();
+            }
+            _ => {
+                panic!("Erase failed: {:?}", error);
+            }
+        }
+    }
+}
+
+impl<A: Alarm<'static>> AlarmClient for LogTest<A> {
+    fn alarm(&self) {
+        self.run();
+    }
+}
+
+impl<A: Alarm<'static>> gpio::Client for LogTest<A> {
+    fn fired(&self) {
+        // Erase log.
+        self.state.set(TestState::Erase);
+        self.erase();
+    }
+}
+
+fn entry_id_to_test_value(entry_id: usize) -> u64 {
+    // Page and entry header sizes for log storage.
+    const PAGE_SIZE: usize = 512;
```

Comment:
This needs to be updated to reflect the nrf52 page size

Reply:
Thanks for catching that!